src/main/java/uk/ac/ebi/atlas/hcalandingpage/HcaHumanExperimentDao.java

```diff
@@ -18,41 +22,46 @@
 import static uk.ac.ebi.atlas.solr.cloud.collections.SingleCellAnalyticsCollectionProxy.ONTOLOGY_ANNOTATION_LABEL;
 
-@Component
+@Repository
 public class HcaHumanExperimentDao {
     private final SingleCellAnalyticsCollectionProxy singleCellAnalyticsCollectionProxy;
 
     public HcaHumanExperimentDao(SolrCloudCollectionProxyFactory solrCloudCollectionProxyFactory) {
         this.singleCellAnalyticsCollectionProxy = solrCloudCollectionProxyFactory.create(SingleCellAnalyticsCollectionProxy.class);
     }
 
-    public ImmutableSet<String> fetchExperimentAccessions(String characteristicName, String characteristicValue) {
+    public ImmutableSet<String> fetchExperimentAccessions(String characteristicName,
+                                                          Set<String> characteristicValues) {
         var queryBuilder = new SolrQueryBuilder<SingleCellAnalyticsCollectionProxy>();
         /* Reason for checking blank here is in case characteristicValue is empty our solrQueryBuilder
          * builds query like characteristic_value:("\*") which is different from characteristic_value:*
          * which we need. So to handle that situation if characteristicValue is empty we build query
          * characteristic_name:("organism_part") as we need all the experiments in case of empty characteristicValue
          * */
-        queryBuilder
-                .setNormalize(false)
-                .addQueryFieldByTerm(CHARACTERISTIC_NAME, characteristicName)
+
+        queryBuilder.addQueryFieldByTerm(CHARACTERISTIC_NAME, characteristicName)
                 .setFieldList(EXPERIMENT_ACCESSION)
-                .setRows(10000000);
-        if (isNotBlank(characteristicValue)) {
-            queryBuilder
-                    .addQueryFieldByTerm(ImmutableMap.of(
-                            CHARACTERISTIC_VALUE, ImmutableList.of("*".concat(characteristicValue)),
-                            ONTOLOGY_ANNOTATION, ImmutableList.of("*".concat(characteristicValue)),
-                            ONTOLOGY_ANNOTATION_ANCESTORS_LABELS, ImmutableList.of("*".concat(characteristicValue)),
-                            ONTOLOGY_ANNOTATION_ANCESTORS_URIS, ImmutableList.of("*".concat(characteristicValue)),
-                            ONTOLOGY_ANNOTATION_LABEL, ImmutableList.of("*".concat(characteristicValue))
-                    ));
+                .sortBy(EXPERIMENT_ACCESSION, SolrQuery.ORDER.asc)
+                .setRows(SolrQueryBuilder.DEFAULT_ROWS);
+
+        if(!characteristicValues.isEmpty()) {
+            queryBuilder.addQueryFieldByTerm(ImmutableMap.of(
+                    CHARACTERISTIC_VALUE, characteristicValues,
+                    ONTOLOGY_ANNOTATION, characteristicValues,
+                    ONTOLOGY_ANNOTATION_ANCESTORS_LABELS, characteristicValues,
+                    ONTOLOGY_ANNOTATION_ANCESTORS_URIS, characteristicValues,
+                    ONTOLOGY_ANNOTATION_LABEL, characteristicValues
+            ));
```

Comment:
as mentioned above we would need the concatenation of `*` in the case when just `ontology_id` is passed

Reply:
Fixed it.